plugins/wdc/wdc-nvme.c

```diff
@@ -992,15 +1018,71 @@ static int wdc_get_pci_ids(uint32_t *device_id, uint32_t *vendor_id)
 	return ret;
 }
 
+static int wdc_get_vendor_id(int fd, uint32_t *vendor_id)
+{
+	int ret;
+	struct nvme_id_ctrl ctrl;
+
+	memset(&ctrl, 0, sizeof(struct nvme_id_ctrl));
+	ret = nvme_identify_ctrl(fd, &ctrl);
+	if (ret) {
+		fprintf(stderr, "ERROR : WDC : nvme_identify_ctrl() failed "
+			"0x%x\n", ret);
+		return -1;
+	}
+
+	*vendor_id = (uint32_t) ctrl.vid;
+
+	return ret;
+}
+
+static bool wdc_check_power_of_2(int num)
+{
+	int first_set = 1;
+
```

Comment:
Better you can have below logic
`return (num && ( !(num & (num-1))));`

Reply:
Jeemon-Joy,
Just to confirm, you're suggesting I change the wdc_check_power_of_2 function to the following, correct?
static bool wdc_check_power_of_2(int num)
{
        return (num && ( !(num & (num-1))));
}

Reply:
Updated the wdc_check_power_of_2 function with your suggested change.